gdx/src/com/badlogic/gdx/scenes/scene2d/ui/TextField.java

```diff
@@ -542,6 +542,16 @@ int delete (boolean fireChangeEvent) {
 		return minIndex;
 	}
 
+	void undo (boolean fireChangeEvent) {
```

Comment:
Is there a reason this isn't public? I would think users might want to perform `undo()` programatically sometimes. I think using package-private by default is a mistake, anyway, for both Java and libGDX, since it doesn't actually provide any kind of prevention from use, it just makes it more cumbersome to use and means people think they don't have to document code...